drivers/sensor/omron_2smpp03/omron_2smpp03.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2020 Teslabs Engineering S.L.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/adc.h>
+#include <drivers/sensor.h>
+#include <devicetree/adc.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(omron_2smpp03, CONFIG_SENSOR_LOG_LEVEL);
+
+/** ADC gain. */
+#define ADC_GAIN ADC_GAIN_1
+/** ADC reference. */
+#define ADC_REF ADC_REF_INTERNAL
```

Comment:
This may not work with all ADCs.

Reply:
I can imagine not all ADCs will have such feature, however I don't know how to handle this situation without adding platform specific code (I think this is what Grove sensor does). IMO ADC needs some API enhancements to avoid situations like this one. Having platform specific code in this type of drivers is a kind of a hack. Any suggestions on how to handle this situation for now?

Reply:
Since I did the Grove example, and I believe the ADC API cannot be made generic, I think it's fine to have the driver work for the devices you've tested on.  When somebody wants to use it on one where these settings don't work there'll have to be peripheral-specific modifications to the driver, just like Grove has.